warehouse/ingest-core/src/test/java/datawave/ingest/data/config/ingest/ErrorShardedIngestHelperTest.java

```diff
@@ -0,0 +1,64 @@
+package datawave.ingest.data.config.ingest;
+
+import java.util.Set;
+
+import org.apache.hadoop.conf.Configuration;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import datawave.TestBaseIngestHelper;
+import datawave.ingest.data.TypeRegistry;
+import datawave.ingest.data.config.DataTypeHelper;
+import datawave.policy.IngestPolicyEnforcer;
+
+class ErrorShardedIngestHelperTest {
+
+    private static final String DATA_TYPE_NAME = "error";
+
+    /**
+     * Verify that when indexed and reversed indexed fields are provided, that they are correctly parsed and are not treated as disallowed fields.
+     */
+    @Test
+    void testSetupGivenIndexedFieldLists() {
+        Configuration config = getBaseConfig();
+        config.set(DATA_TYPE_NAME + BaseIngestHelper.INDEX_FIELDS, "FOO,BAR,HAT");
+        config.set(DATA_TYPE_NAME + BaseIngestHelper.REVERSE_INDEX_FIELDS, "APPLE,BANANA,KIWI");
+
+        ErrorShardedIngestHelper helper = new ErrorShardedIngestHelper();
+        helper.setup(config);
+
+        Assertions.assertEquals(Set.of("FOO", "BAR", "HAT"), helper.getIndexedFields());
+        Assertions.assertFalse(helper.hasIndexDisallowlist());
+
+        Assertions.assertEquals(Set.of("APPLE", "BANANA", "KIWI"), helper.getReverseIndexedFields());
+        Assertions.assertFalse(helper.hasReverseIndexDisallowlist());
+    }
+
+    /**
+     * Verify that when disallowed indexed and reversed indexed fields are provided, that they are correctly parsed and are treated as disallowed fields.
+     */
+    @Test
+    void testSetupGivenDisallowedIndexedFieldLists() {
+        Configuration config = getBaseConfig();
+        config.set(DATA_TYPE_NAME + BaseIngestHelper.DISALLOWLIST_INDEX_FIELDS, "FOO,BAR,HAT");
+        config.set(DATA_TYPE_NAME + BaseIngestHelper.DISALLOWLIST_REVERSE_INDEX_FIELDS, "APPLE,BANANA,KIWI");
+
+        ErrorShardedIngestHelper helper = new ErrorShardedIngestHelper();
+        helper.setup(config);
+
+        Assertions.assertEquals(Set.of("FOO", "BAR", "HAT"), helper.getIndexedFields());
```

Comment:
I was deeply confused when I read this, but now I see what's going on and I'm sorry we did this to you, and to ourselves.  

It looks like when we have a disallowed list of index fields, we still populate the indexedFields Set in the BaseIngestHelper.  isIndexedField will check the hasIndexDisallowlist and negate the result.  However, getIndexedFields is still public and anything accessing that would need to know to check the disallow flag.  Jimmies.  Rustled.

But, as none of that is your fault, I think it would be less heart-attack inducing to see the an assertion that FOO, BAR, etc. return false from the isIndexedField method until we can get the aforementioned hokiness resolved via a different ticket,